sympy/solvers/solveset.py

```diff
@@ -847,6 +847,9 @@ def solveset(f, symbol=None, domain=S.Complexes):
 
     if not free_symbols:
         b = Eq(f, 0)
+        if isinstance(symbol, Number):
```

Comment:
Yes, if I replace the `if condition` with `getattr(symbol, 'is_Symbol', False)` then the test case:
`solveset(0, domain=S.Reals)` won't pass. It should return `S.Reals` but it raises `ValueError` instead.
I guess it is due to the fact that the case in [LINE](https://github.com/sympy/sympy/blob/master/sympy/solvers/solveset.py#L837):
```
if f is S.true:       # returns False
   return domain
``` 
therefore the condition is not checked and it goes to line 848 and raises a `ValueError`